src/test/java/org/folio/dcb/listener/CirculationLoanEventListenerTest.java

```diff
@@ -155,6 +170,75 @@ void handleCheckingInWithIncorrectDataTest() {
+
+    verify(baseLibraryService, times(1)).updateTransactionEntity(transactionEntityArgumentCaptor.capture(),
+            statusEnumArgumentCaptor.capture());
+    TransactionEntity transactionEntityArgumentCaptorValue = transactionEntityArgumentCaptor.getValue();
+    assertEquals(BORROWING_PICKUP, transactionEntityArgumentCaptorValue.getRole());
+    assertEquals(Boolean.TRUE, transactionEntityArgumentCaptorValue.getSelfBorrowing());
+    assertEquals(TransactionStatus.StatusEnum.ITEM_CHECKED_OUT, statusEnumArgumentCaptor.getValue());
+
+  }
+
+  @Test
+  void handleCheckOutEventInBorrowingPickupSelfBorrowingFalseCheckout_negative() {
+    var transactionEntity = createTransactionEntity();
+    transactionEntity.setItemId("8db107f5-12aa-479f-9c07-39e7c9cf2e4d");
+    transactionEntity.setStatus(TransactionStatus.StatusEnum.CREATED);
+    transactionEntity.setRole(BORROWING_PICKUP);
+    transactionEntity.setSelfBorrowing(Boolean.FALSE);
+    MessageHeaders messageHeaders = getMessageHeaders();
+    when(transactionRepository.findSingleTransactionsByItemIdAndStatusNotInClosed(any())).thenReturn(Optional.of(transactionEntity));
+    eventListener.handleLoanEvent(CHECK_OUT_EVENT_SAMPLE, messageHeaders);
+    verify(transactionRepository, times(0)).save(any());
+  }
+
+
+  @Test
+  void handleCheckOutEventInBorrowingPickupSelfBorrowingTrueCheckin_positive() {
+    var transactionEntity = createTransactionEntity();
+    transactionEntity.setItemId("8db107f5-12aa-479f-9c07-39e7c9cf2e4d");
+    transactionEntity.setStatus(TransactionStatus.StatusEnum.CREATED);
+    transactionEntity.setRole(BORROWING_PICKUP);
+    transactionEntity.setSelfBorrowing(Boolean.TRUE);
+    MessageHeaders messageHeaders = getMessageHeaders();
+    when(transactionRepository.findSingleTransactionsByItemIdAndStatusNotInClosed(any())).thenReturn(Optional.of(transactionEntity));
+    eventListener.handleLoanEvent(CHECK_IN_DCB_FALSE_LOAN_EVENT_SAMPLE, messageHeaders);
+    verify(transactionRepository, times(1)).save(any());
+
+    verify(baseLibraryService, times(1)).updateTransactionEntity(transactionEntityArgumentCaptor.capture(),
+            statusEnumArgumentCaptor.capture());
+    TransactionEntity transactionEntityArgumentCaptorValue = transactionEntityArgumentCaptor.getValue();
+    assertEquals(BORROWING_PICKUP, transactionEntityArgumentCaptorValue.getRole());
+    assertEquals(Boolean.TRUE, transactionEntityArgumentCaptorValue.getSelfBorrowing());
+    assertEquals(TransactionStatus.StatusEnum.CLOSED, statusEnumArgumentCaptor.getValue());
+  }
+
+  @Test
+  void handleCheckOutEventInBorrowingPickupSelfBorrowingTrueCheckin_negative() {
+    var transactionEntity = createTransactionEntity();
+    transactionEntity.setItemId("8db107f5-12aa-479f-9c07-39e7c9cf2e4d");
+    transactionEntity.setStatus(TransactionStatus.StatusEnum.CREATED);
+    transactionEntity.setRole(BORROWING_PICKUP);
+    transactionEntity.setSelfBorrowing(Boolean.FALSE);
```

Comment:
It's not critical, but such construction can be extracted to a static method, and variable fields can be passed as method arguments, which will increase readability